src/Illuminate/Http/Middleware/LocalizeRequest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Illuminate\Http\Middleware;
+
+use Closure;
+use Illuminate\Contracts\Foundation\Application;
+
+class LocalizeRequest
+{
+    /**
+     * The application instance.
+     *
+     * @var \Illuminate\Contracts\Foundation\Application
+     */
+    protected $app;
+
+    /**
+     * Create a new Middleware instance.
+     *
+     * @param  \Illuminate\Contracts\Foundation\Application  $app
+     * @return void
+     */
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  \Closure $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $guessedLocale = $request->segment(1);
```

Comment:
Why not make this into a separate method, so we can extends on just override this as we needed.

Something like: https://github.com/orchestral/translation/blob/master/src/Middleware/Locale.php

Implementation example: https://github.com/orchestral/translation/blob/master/src/Middleware/BrowserLocale.php (check against browser prefered language).